packages/flutter/lib/src/widgets/scrollbar.dart

```diff
@@ -437,38 +437,55 @@ class ScrollbarPainter extends ChangeNotifier implements CustomPainter {
         trackSize = Size(thickness + 2 * crossAxisMargin, _trackExtent);
         x = crossAxisMargin + padding.left;
         y = _thumbOffset;
-        trackOffset = Offset(x - crossAxisMargin, 0.0);
+        trackOffset = Offset(x - crossAxisMargin, mainAxisMargin);
         break;
       case ScrollbarOrientation.right:
         thumbSize = Size(thickness, thumbExtent);
         trackSize = Size(thickness + 2 * crossAxisMargin, _trackExtent);
         x = size.width - thickness - crossAxisMargin - padding.right;
         y = _thumbOffset;
-        trackOffset = Offset(x - crossAxisMargin, 0.0);
+        trackOffset = Offset(x - crossAxisMargin, mainAxisMargin);
         break;
       case ScrollbarOrientation.top:
         thumbSize = Size(thumbExtent, thickness);
         trackSize = Size(_trackExtent, thickness + 2 * crossAxisMargin);
         x = _thumbOffset;
         y = crossAxisMargin + padding.top;
-        trackOffset = Offset(0.0, y - crossAxisMargin);
+        trackOffset = Offset(mainAxisMargin, y - crossAxisMargin);
         break;
       case ScrollbarOrientation.bottom:
         thumbSize = Size(thumbExtent, thickness);
         trackSize = Size(_trackExtent, thickness + 2 * crossAxisMargin);
         x = _thumbOffset;
         y = size.height - thickness - crossAxisMargin - padding.bottom;
-        trackOffset = Offset(0.0, y - crossAxisMargin);
+        trackOffset = Offset(mainAxisMargin, y - crossAxisMargin);
         break;
     }
 
     _trackRect = trackOffset & trackSize;
+    Offset borderStart;
+    Offset borderEnd;
+    switch(resolvedOrientation) {
```

Comment:
Can these not be included in the switch statements above, rather than repeating?